frontend/model/contracts/identity.js

```diff
@@ -121,15 +123,22 @@ sbp('chelonia/defineContract', {
     'gi.contracts/identity/createDirectMessage': {
       validate: (data, { state, getters }) => {
         objectOf({
-          privacyLevel: unionOf(...Object.values(CHATROOM_PRIVACY_LEVEL).map(v => literalOf(v))),
-          contractID: string
+          // NOTE: 'groupContractID' is the contract ID of the group where the direct messages is created
+          //       it's optional parameter meaning the direct message could be created outside of the group
+          groupContractID: optional(string),
+          contractID: string // NOTE: chatroom contract id
         })(data)
       },
       process ({ data }, { state }) {
-        Vue.set(state.chatRooms, data.contractID, {
-          privacyLevel: data.privacyLevel,
-          hidden: false // NOTE: this attr is used to hide/show direct message
-        })
+        const { groupContractID, contractID } = data
+        if (groupContractID) {
+          Vue.set(state.chatRooms, contractID, {
+            groupContractID,
+            visible: true // NOTE: this attr is used to hide/show direct message
+          })
+        } else {
+          // TODO: create a direct message outside of the group
+        }
```

Comment:
To be honest, I have never thought before how to save the DMs from outside of any groups.
Just having thought about it, it could be fine to save the two types of DMs in `state.chatRooms`.
If the `groupContractID` is undefined, it means the DMs outside of the group. I will remove `if/else` statement.